lib/round.rb

```diff
@@ -0,0 +1,96 @@
+require './lib/deck'
+
+class Round
+  attr_reader :deck,
+              :turns
+  attr_accessor :correct_answers,
+                :turn_count,
+                :total_cards,
+                :categories
+
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+    @correct_answers = 0
+    @turn_count = 1
+    @total_cards = 0
+    @categories = []
+  end
+
+  def current_card
+    @deck.cards.first
+  end
+
+  def take_turn(answer, card = current_card)
+    @deck.cards.shift
+    @turn_count += 1
+    new_turn = Turn.new(answer, card)
+    @turns << new_turn
+    @correct_answers += 1 if @turns.last.correct?
+    new_turn
+  end
+
+  def number_correct
+    @correct_answers
+  end
+
+  def percent_correct
+    (@correct_answers.to_f / turns.count) * 100
+  end
+
+  def number_correct_by_category(cat)
+    num_by_cat = turns.count do |turn|
+      turn.correct? && turn.card.category == cat
+    end
+  end
```

Comment:
Nice work here. You don't necessarily need to save the return value of the enumerable into `num_by_cat`, because it will automatically get returned from this method.